### ISSUE TITLE
Loud high-pitch whistling when start playing some MP3 files

### ISSUE THREAD

**Issue by EvgenKo423** [Nevcairiel/LAVFilters#72]
Just installed a K-Lite Codec Pack 12.01 and I have an issue playing an MP3-file through in-game player of some game.

So, I have two types of MP3 files:
* [First](https://www.dropbox.com/s/yhvs4p3qcuyhf1n/test_file1.mp3?dl=1):
```
  Audio
  Format                      : MPEG Audio
  Format version              : Version 2
  Format profile              : Layer 3
```
* [Second](https://www.dropbox.com/s/bacdfgzsf42lihu/test_file2.mp3?dl=1):
```
  Audio
  Format                      : MPEG Audio
  Format version              : Version 1
  Format profile              : Layer 3
```
When I play both of them through MPC, the LAV Audio Decoder and ffdshow Audio Decoder are used and LAV is saying that input codec is `mp3float`. When I play both of them through the in-game music player, only the LAV Audio Decoder is used and is saying that input codec is `mp1float`.

When I play both of them through MPC, no issues happen. When I play the first one through the in-game music player, also no issues happen. But when I play the __second__ one through the in-game music player, I hear loud high-pitch whistling for about half a second.
If I go to LAV Audio Decoder's _Properties -> Formats_ and disable it for `mp2` and `mp3` file types &mdash; no issues happen anymore.

Also, I've found that [some people had been experiencing similar issue with another game](https://steamcommunity.com/app/2820/discussions/0/882962960932510082/) back in October, 2012.

**Comment by Nevcairiel** [comment]
There is nothing I can do, those games just use DirectShow badly. Any proper DirectShow player clearly handles the files properly, as evident by MPC playing them properly.

**Comment by Nevcairiel** [comment]
To be precise, such games open mp3 files with the Microsoft MPEG-I file reader, which tries to read the ID3 tag from the start of the file - but it fails at its job, and only reads half of it, leaving the other half in place. This partial ID3 tag is then fed to LAV Audio, which it tries to decode as audio, because hey, why would a source send me something other than audio, right?

Well, no such luck, its garbage, and garbage input results in garbage audio output.

**Comment by Nevcairiel** [comment]
I had an idea on how to try to skip the garbage at the beginning, will have to see if it actually works the way I want it to.

**Comment by EvgenKo423** [comment]
:grinning: OK, thanks for the clarification anyway.

Maybe there will be a stupid questions, but I thought earlier this day: OK, maybe it is MPEG-I Stream Splitter, but there is no foreign sounds when the data is passed to the MP3 Decoder DMO (instead of LAV), so why it doesn't work with LAV? They're all fed with the same data. MS have a fix for their own bug?

Also, AFAIK, MP3 is a chunked format, and tags and audio stream are different chunks. How the tag data can be passed to the decoder at all? It should be read from the separate data chunks. Or a decoder is also a parser, not just a straight decoder? Just can't find any info on this topic at all...

**Comment by clsid2** [comment]
Which game is it?

You can solve the problem by blacklisting the game:
HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\LAV\Audio\Blacklist

**Comment by EvgenKo423** [comment]
Euro Truck Simulator 2

Actually it will be cool to have a visual editor for black-/whitelist, just like `ffdshow`, because I can't find it myself without your note.

**Comment by Nevcairiel** [comment]
Fixed in 2f112a2698d70d522839f8a188403144bb70ce9f
